tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -56,14 +56,22 @@ def montecarlo_main_loop(packet_collection, numba_model, numba_plasma,
     for i in prange(len(output_nus)):
         if montecarlo_configuration.single_packet_seed != -1:
             i = montecarlo_configuration.single_packet_seed
-        r_packet = RPacket(numba_model.r_inner[0],
-                           packet_collection.packets_input_mu[i],
-                           packet_collection.packets_input_nu[i],
-                           packet_collection.packets_input_energy[i],
-                           i)
-
-        # We want to set the seed correctly per user; otherwise, random.
-        np.random.seed(i)
+            r_packet = RPacket(numba_model.r_inner[0],
+                               packet_collection.packets_input_mu[i],
+                               packet_collection.packets_input_nu[i],
+                               packet_collection.packets_input_energy[i],
+                               i)
+            np.random.seed(i)
```

Comment:
I am a little bit lost here. As far as I can tell, the purpose of the `single_packet_seed` is to re-run a packet for debugging purposes. The index of this original packet is the single_packet_seed. So why do we not use the same seed as for the original packet (i.e. `seed = packet_seeds[i]`)? However, my interpretation of the purpose and implementation of the `single_packet_seed` might be completely wrong.

Reply:
This way, we can set `single_packet_seed` to be whatever we'd like — so if we run a simulation and everything works well except for packet 725, we can just input that packet number directly and run a simulation without any other packets.